### PR TITLE
Disable built-in functionalization when we use functorch's functionalization

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #568

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: torchdynamo/optimizations/normalize.py [hunk 1]
```diff
@@ -429,8 +429,11 @@ def normalize_ir(gm, example_inputs):
         except AttributeError:
             # log.exception("NormalizeOperators() failed")
             pass
-        ShapeAliasingAndMutationProp(gm).run(*example_inputs)
-        gm = Functionalization(gm).transform()
+        import functorch._src.config
+
+        if not functorch._src.config.use_functionalize:
```
Comment (jansel) [inline torchdynamo/optimizations/normalize.py:434]: I think this check should go here instead:
https://github.com/pytorch/torchdynamo/blob/bb60ba951dcccb8de5aef6e930ef6cc7a30b24e3/torchdynamo/optimizations/training.py#L32

So it only affect AOT Autograd (and not other backends).

**Review by jansel — CHANGES_REQUESTED** [review]

**Comment by anijain2305** [comment]
Duplicate - https://github.com/pytorch/torchdynamo/pull/593. That PR is basically Jason's suggestion.

**Comment by ezyang** [comment]
closing in favor of @anijain2305's version
